### ISSUE TITLE
Questions about the G(z) in the paper and the code

### ISSUE THREAD

**Issue by DCMMC** [SPIN-UMass/BLANKET#1]
First of all thank you very much for your work. We are pleased to be able to read your paper. However, there are some questions when we reading the paper and the code:

* What is the **detailed** formula of `G(z)`, especially how `x` and `z` are linked in `G`? According to the code, we guess `G` is defined as `ReLU(FC1(FC2(x)))` where `FC1` and `FC2` are fully connected layers. But `z` is not appeared in the code.
*  Would you be willing to generously share the code for the training? Specifically, how the training process addresses the challenge of **live** network traffic. According to the code, we guess history window is applied to ensure this demand.

We would be very appreciated if you can help us. Thank you very much!

Best wishes.

**Comment by pkarthik88** [comment]
Thanks for the well-written paper and releasing the code. I can confirm that I found it difficult to locate the code corresponding to the training phase. This is particularly important since the paper suggests multiple gradients and a special case where the function is not differentiable. 

Another aspect where I require some help is relating the mapping function (domain specific constraints are specified as mapping functions) to the codebase. 

Looking forward to your response. Thanks.

**Comment by srxzr** [comment]
I uploaded the code for the training of the generator function. You need to download the deepcorr training data. Or you can use it on your own dataset.

**Comment by Zentions** [comment]
Thanks for the excellent paper and code.  We are now doing experiments on website fingerprint defense. Would you like to share the training code on website fingerprint.

Looking forward to your reply soon!  Thank you very much!
Best wishes.

**Comment by lpf-2001** [comment]
This paper is based on the DeepCorr classification model. For the training code of the classification model, you can find it here: https://github.com/SPIN-UMass/DeepCorr.
